.github/workflows/build_lint.yml

```diff
@@ -60,18 +60,23 @@ jobs:
       with:
         path: ${{ env.pythonLocation }}
         key: ${{ runner.os }}-pip-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}
+    # Workaround; remove once https://github.com/astral-sh/uv/issues/1386 is fixed
+    - run: echo "VIRTUAL_ENV=${Python_ROOT_DIR}" >> $GITHUB_ENV
+    - name: Install uv
+      run: pip install uv
     - name: Install dependencies
       # Only if the cache misses
       # Based on https://github.com/pypa/pip/issues/8049#issuecomment-633845028
       # read_requirements.py should be removed once
       # https://github.com/pypa/pip/issues/11440 is resolved.
       if: steps.cache.outputs.cache-hit != 'true'
       run: |
-        pip install toml
+        rm -fr ${Python_ROOT_DIR}/lib/python${{ matrix.python-version}}/site-packages/pip-23.0.1.dist-info
```

Comment:
Why is this line needed? Seems like it would break easily when the pip version changes.

Reply:
`uv` breaks if there are multiple `pip` installed. There isn't a bug report for this yet, though if I were to open one, would be resolved within days because of the use case in GH Actions.

Reply:
Are you sure about this one? Specifically I don't see how why multiple pips should be installed at all and why uv should interact with them at all. Also I haven't seen this in other workflow files that already use `uv`

Reply:
That's because they use `uv venv`, e.g. https://github.com/astral-sh/uv/issues/1386#issuecomment-1952420541.

Reply:
I was thinking of waiting until the non-venv solution has stabilized, but `uv venv` is blazing fast, so it shouldn't matter to create the venv.

Reply:
`uv venv` failed https://github.com/projectmesa/mesa/actions/runs/7980718647/job/21790983916?pr=2038 because Solara has a hardcoded assumption on the virtualenv information.